### ISSUE TITLE
Idea: Unique Buildings (and possible gameplay plot arc)

### ISSUE THREAD

**Issue by baxil** [blackhole12/PonyClicker#75]
I suggested this in issue #67, but I'm pulling it out and deleting it there because it should be its own thing.

Perhaps for a later milestone, create new upgrades -- or a new class of upgrades -- that are different unique Ponyville buildings? The most basic could have effects related to your wheel.

Start with only Pinkie Pie and background ponies unlocked for the wheel.
Train Station - When purchasing a new pony, you may select which one you buy.
Sweet Apple Acres - Unlocks Granny Smith, Applejack, Apple Bloom, Big Mac
Carousel Boutique - Unlocks Rarity, Sweetie Belle
Napping Cloud - Unlocks Rainbow Dash, Scootaloo
Gate to Tartarus - Unlocks Chrysalis, Sombra, other villains
The Friendship Castle - Unlocks alicorns
The Everfree Forest - Unlocks Fluttershy, and ...

... ooh, I just realized this might tie in with the Mirrorpocalypse: maybe you need to unlock the Everfree to be able to explore it and find late-game game-overhaul features like the Pool. That suggests that maybe (some or all) buildings are rewards for resetting after hitting certain muffin targets (or gameplay-time targets)?

This also suggests a possible actual plotline and endgame state beyond simply "earn smiles, get ponies". Ponyville has fallen through a Sadness Vortex (or insert other plot-coupon quest here) and your goal is, with the power of smiles (and muffins), to restore everything back to the way it was. You can have a win state when you buy a certain building (Sugarcube Corner) or when Ponyville is fully restored, and then have that achievable without 100%ing the game so that completionists can continue on past it to complete, and the truly obsessed can just keep going indefinitely in search of smile records.

Other buildings maybe could have effects other than unlocking ponies, but I don't know what yet. Sugarcube Corner, Town Hall, The Library Tree, and Quills and Sofas seem like obvious candidates. Maybe they unlock upgrades, or they're late-game bonuses that are kick-plot upgrades in themselves, or they give you new clicking modes or otherwise freshen the game as the curves are starting to level off. (The Mirror Pool then becomes the secret crazy ultra bonus level.)

Just to add, I really like the idea of an idle clicker game with an actual plot and win state.  I don't think that's actually been done; at any rate, the only ones I've ever played have all been "keep going until you run out of achievements".

**Comment by blackhole12** [comment]
I like the general idea here and I think it synergies well with the suggested UI overhaul in #50. However, it's obvious that this would be a significant time investment to implement and require even more advanced analysis and graphs to ensure it is properly balanced. This may require nuking the pony cost curve by setting it on an n^0.9 curve or something, because otherwise the resulting price explosion makes any ponies past 28 almost completely impossible.

**Comment by baxil** [comment]
To be clear, I'm not suggesting that purchasing buildings adds ponies to the wheel.  I'm suggesting that it adds to the pool of available ponies that clicking "Buy a Pony" might randomly add to your wheel.  The +pony buildings would be *cosmetic* upgrades.

**Comment by Abion47** [comment]
I think that, as the game currently stands, this would have to be an extremely late game thing. In my game, I've gotten to the point where I can unlock every upgrade and get nonillions to decillions of SPS within 5 minutes of a Reset, and I still haven't really been able to have half the maximum number of ponies I can get in a single session.

Either that, or overhaul the current system of how ponies are generated (which would include a changing maximum of ponies, affect the SPS curve of the game, and potentially screw with existing saves).

**Comment by ILM126** [comment]
Yeah, this would be an amazing addition to this game. I've been playing Adventure Capitalist for the past few weeks, only about 20 min every week.. But I noticed a pattern in the game, almost every item is a 3x more than the previous item. But maybe I'll bring this up on another issue.

But also, the game had a rough storyline.

**Comment by Sailorleo** [comment]
This should probably be used as the core for a sequel.  Trying to implement it in the current game would require redoing almost everything (production would be too slow in the early game because of the limits on available ponies; if we don't change cost curves for ponies and friendships the cost of ponies is too high in late game to actually get the newly unlocked ponies and the production values for most buildings only work well for the current costs).
